src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,69 +1,81 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import IconQuestion from "@elements/Icon/Generated/IconQuestion";
+import React, { cloneElement, FC, ReactElement, ReactNode } from "react";
+import { Button, ButtonStyle, ButtonSize } from "@elements/Button/Button";
 import { IconSize } from "@elements/Icon/IconSize";
-import { useFocusVisible } from "@react-aria/interactions";
-import { useTooltip, useTooltipTrigger } from "@react-aria/tooltip";
-import { useTooltipTriggerState } from "@react-stately/tooltip";
-import { FOCUS_STYLE } from "@utilities/focusStyle";
-import { merge } from "@utilities/merge";
-import React, { FC, ReactNode, useRef, useState } from "react";
-import { usePopper } from "react-popper";
+import { BrightHeader, BrightHeaderVariants } from "./partials/BrightHeader";
+
+export type TooltipButton = {
+    label: string;
+    action: () => void;
+};
 
 export type TooltipProps = {
     tooltip: ReactNode;
+    tooltipIcon?: ReactElement;
+    heading?: ReactNode;
+    headingIcon?: ReactElement;
+    linkUrl?: string;
+    linkLabel?: string;
+    brightHeader?: BrightHeaderVariants;
+    buttons?: [TooltipButton, TooltipButton];
 };
 
-const TOOLTIP_DISTANCE = 9;
-const TOOLTIP_SKIDDING = 0;
-
-export const Tooltip: FC<TooltipProps> = ({ tooltip }) => {
-    const [tooltipTriggerElement, setTooltipTriggerElement] = useState<HTMLElement | null>(null);
-    const [tooltipElement, setTooltipElement] = useState<HTMLDivElement | null>(null);
-    const state = useTooltipTriggerState();
-    const { triggerProps, tooltipProps } = useTooltipTrigger({}, state, useRef(tooltipTriggerElement));
-    const { tooltipProps: tooltipAriaProps } = useTooltip(tooltipProps, state);
-    const { isOpen } = state;
-    const { isFocusVisible } = useFocusVisible();
-    const { styles, attributes } = usePopper(tooltipTriggerElement, tooltipElement, {
-        placement: "auto-end",
-        modifiers: [
-            {
-                name: "offset",
-                options: {
-                    offset: [TOOLTIP_SKIDDING, TOOLTIP_DISTANCE],
-                },
-            },
-        ],
-    });
+export const Tooltip: FC<TooltipProps> = ({
+    tooltip,
+    tooltipIcon,
+    heading,
+    headingIcon,
+    linkUrl,
+    linkLabel,
+    brightHeader,
+    buttons,
+}) => {
+    const showButtons = buttons !== undefined && Object.keys(buttons).length === 2;
```

Comment:
```suggestion
    buttons = [],
}) => {
    const showButtons = Object.keys(buttons).length === 2;
```

It can only be `undefined | [TooltipButton, TooltipButton]`
`undefined` is falsy and `[TooltipButton, TooltipButton]` is truthy